myhdl/conversion/_VHDLNameValidation.py

```diff
@@ -33,16 +34,19 @@
 # ensure reserved words are not being used for the wrong purpose
 def _nameValid(name):
     if name.lower() in _vhdl_keywords:
-        warnings.warn("VHDL keyword used: %s" % (name))
+        warnings.warn("VHDL keyword used: {}".format(name), category=ToVHDLWarning)
 
     if name.startswith('_'):
-        warnings.warn("VHDL variable names cannot start with '_': %s" % (name))
+        warnings.warn("VHDL variable names cannot start with '_': {}".format(name), category=ToVHDLWarning)
 
     if '-' in name:
-        warnings.warn("VHDL variable names cannot contain '-': %s" % (name))
+        warnings.warn("VHDL variable names cannot contain '-': {}".format(name), category=ToVHDLWarning)
+
+    if '__' in name:
+        warnings.warn("VHDL variable names cannot contain double underscores '__': {}".format(name), category=ToVHDLWarning)
```

Comment:
We should have a test for this too.

Reply:
@josyb tells me this test has been added.